File: client/api/models/recipe.ts
import {Nutrients} from "@/api/models/nutrition";

export interface Recipe {
    // string `json:"category"`
    category: string;

    // int `json:"cook_time"`
    cookTime: number;

    // string `json:"description"`
    description: string;

    // string `json:"image"`
    image: string;

    // []string `json:"ingredients"`
    ingredients: string[];

    // string `json:"instructions"`
    instructions: string;

    // []string `json:"keywords"`
    keyWords: string[];

    // Nutrients `json:"nutrients"`
    nutrients: Nutrients; // Reference to the Nutrients interface

    // int `json:"prep_time"`
    prepTime: number;

    // float64 `json:"ratings"`
    ratings: number;

    // int `json:"total_time"`
    totalTime: number;

    // string `json:"title"`
    title: string;

    // string `json:"yields"`
    yields: string;
}

File: backend/core/src/main/java/decidish/com/core/model/recipes/Recipe.java
package decidish.com.core.model.recipes;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;
import java.util.ArrayList;

@Entity
@Table(name = "recipes")
@Getter @Setter
@NoArgsConstructor
@AllArgsConstructor
public class Recipe {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true)
    private String title;

    // @Column(columnDefinition = "TEXT")
    // private String description;

    // @Column(columnDefinition = "TEXT")
    // private String instructions;

    private Integer cookTime;
    private Integer prepTime;
    private Integer totalTime;

    private String image;
    private Float rating;
    private String servingSize;
    private String calories;

    @OneToMany(
        mappedBy = "recipe",
        cascade = CascadeType.ALL,
        orphanRemoval = true
    )
    private List<RecipeIngredient> ingredients = new ArrayList<>();
}

